lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -1056,6 +1056,40 @@ void LayoutRematerialization::hoistConvertIntoConditionals() {
   }
 }
 
+static bool isExpensiveMathOp(Operation *op) {
+  // These operations are either multiple instructions or have throughput
+  // lower than 16 according to the arithmetic instructions table in:
+  // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#arithmetic-instructions
+  static const llvm::DenseSet<llvm::StringRef> expensiveOps = {
+      "arith.divf", "math.erfc",  "math.sinh",  "math.cosh",  "math.tanh",
+      "math.asinh", "math.acosh", "math.atanh", "math.ctlz",  "math.cttz",
```

Comment:
nit: can we use `math::SinOp::getOperationName` just in case the names change in the future?